crates/sui-move-build/src/lib.rs

```diff
@@ -223,7 +224,7 @@ pub fn build_from_resolution_graph(
     run_bytecode_verifier: bool,
     print_diags_to_stderr: bool,
 ) -> SuiResult<CompiledPackage> {
-    let (published_at, dependency_ids) = gather_published_ids(&resolution_graph);
+    let (published_at, dependency_ids) = gather_published_ids(&resolution_graph, None);
```

Comment:
A `move build` or `publish` command does not yet try to resolve from the `Move.lock`, therefore `None` here.